src/dao/index.js

```diff
@@ -0,0 +1,17 @@
+const config = require('~/src/config')
+const logger = require('~/src/logging').logger(module)
+const FashionKnex = require('windbreaker-service-util/dao/FashionKnex')
+
+// TODO: move this base implementation class to shared utils?
+class BaseDao extends FashionKnex {
```

Comment:
This approach is a little different from the dao pattern in hooks. I needed a way to extend common functionality for each dao/table and opted for this over using the `createDaoHelper`.

Reply:
It might be better to hide the underlying implementation, so that if we ever want to swap/change it in utils, it will be changed everywhere. Thoughts on just overriding the `createDaoHelper` function?

```js
const createDaoUtil = require('windbreaker-service-util/dao')
const config = require('~/src/config')
const logger = require('~/src/logging').logger(module)

module.exports = function createDaoHelper (options) {
  return createDaoUtil({
    ...options,
    logger,
    knexConfig: config.getKnex()
  })
}
```

Reply:
But I agree that ultimately this functionality should be overwritten since you're using it in multiple places.

Reply:
I went with this approach because it naturally felt like inheritance - there's some base behavior, and we want to extend it per dao type. Currently, there isn't much common functionality here but thought there might be eventually. I originally extended BaseDao for each specific Dao, but it started to trend toward a singleton pattern and I'm not a huge fan of singletons. It does seem odd to use a BaseDao and not extend it, though. I could be convinced either way.